### DISCUSSION TITLE
plotting video/image in plotjuggler?

### DISCUSSION THREAD

**Discussion by TomerSegalAB** [facontidavide/PlotJuggler#962] [Q&A]
hi,

- is it possible to plot a ros2 `sensor_msgs/Image` or  `sensor_msgs/CompressedImage` in plotjuggler? 
- i see a  [VideoViewer](https://github.com/facontidavide/PlotJuggler/tree/main/plotjuggler_plugins/VideoViewer) folder in the code, where can i find any documentation about using it?

thanks

**Comment by h4tr3d** [comment]
VideoViewer plugin can load any video. To link internal video (frame) timestamps required some dataset in the plotjuggler, that maps timestamp to the frame number. This dataset must be dragged into reference dataset field in the VideoViewer plugin after that video will be synced with time slider in the main window.

We written our own plugin that loads our MKV (Matroska) files with per-frame metadata (via BlockAdditionals). Frames exported as a any sequence and can be loaded by the our Viewer plugin (based on VideoViewer). special metadata tag `pts` contains time stamp in the same time domain that used by the main Ulog log. So, we load MKV file as a generic datasource, but after loading just "merge" it with currently loaded dataset.

**Comment by h4tr3d** [comment]
> is it possible to plot a ros2 sensor_msgs/Image or sensor_msgs/CompressedImage in plotjuggler?

I am not sure, if this messages properly handled by the appropriate DataLoader plugin. But, plugin architecture provide DataParser plugins. May be you should write your own one and export `Image` / `CompressedImage` to the `any` dataset. After that write StatePublisher plugin that can draw your sequence.

**Comment by facontidavide** [comment]
FYI, video and image support is in the roadmap and will be dramatically improved in the next months

**Comment by h4tr3d** [comment]
> FYI, video and image support is in the roadmap and will be dramatically improved in the next months

Will wait to look into!
